challenge3/webapp/yoshikawa_keishi/spec/models/house_spec.rb

```diff
@@ -0,0 +1,51 @@
+require 'rails_helper'
+describe House do
+  describe '#create' do
+    it "can be saved with all columns filled" do
+      house = build(:house)
+      expect(house).to be_valid
+    end
+
+    it "can't be saved without csv_id" do
+      house = build(:house, csv_id: nil)
+      house.valid?
+      expect(house.errors[:csv_id]).to include("can't be blank")
```

Comment:
includeマッチャだと、それ以外のエラーが入っていてもテストが通ってしまうので、contain_exactlyなどのマッチャを使うと良いかと思います。

Reply:
今日仰っていたところですね。下のfull_nameのテストも含めて週末トライしてみます！

Reply:
このようにしていけました！contain_exactlyにするまでは他の配列要素の存在を見過ごしいました。ありがとうございます^^;
```suggestion
      expect(house.errors[:csv_id]).to contain_exactly("can't be blank", "is not a number")
```

Reply:
例えば、blankの場合は、blankのメッセージだけ出して、blankじゃない場合は、numberかどうかのチェックを走らせるといったvalidationを実装してみるとちょっと面白いかもしれません！

Reply:
このように、２行に分けて、２行目をcsv_idがblankじゃない時と条件分けしたらできました！
```
  validates :csv_id, presence: true
  validates :csv_id, numericality: { only_integer: true }, unless: Proc.new { |a| a.csv_id.blank? }
```
しかしこの書き方だと行数が結構増えるので、美しくない気もしています。
またnumericalityで指定した場合は、わざわざpresence: trueを指定しなくても、blankを受け付けないという機能は果たせるので、今回はpresence: trueを書く必要がそもそもいらない気もしてきました。

Reply:
実装的にはそんな感じかと思います。
強いていうなら、unless blank?が二重否定になってしまってわかりにくいので、
```ruby
validates :csv_id, numericality: { only_integer: true }, if: -> { csv_id.present? }
```
としちゃった方がいいかもしれません。。

そうですね。ただ、例えば、フォームで入力した場合のエラーを考えた時に、未入力時に、
「csv_idは数値で入力してください」
とエラーを出すのか、まずは未入力なんだから、
「csv_idを入力してください」
とエラーを出すのかは仕様によります。numericalityだけだと、「csv_idは数値で入力してください」が出てしまうと思います。
どちらの場合でも実装できるようにしておくと良いでしょう。

Reply:
そんなにシンプルに書けたんですね(°_°)
いずれの場合にも実装できるようにします！

Reply:
ありがとうございます！